pkg/label/label.go

```diff
@@ -317,74 +314,82 @@ func (l Label) Restore(val string) Label {
 
 // PD assigns pd to component key in label
 func (l Label) PD() Label {
-	l.Component(PDLabelVal)
-	return l
+	return l.Component(PDLabelVal)
+}
+
+// IsPD returns whether label is a PD
+func (l Label) IsPD() bool {
+	return l[ComponentLabelKey] == PDLabelVal
 }
 
 // Pump assigns pump to component key in label
 func (l Label) Pump() Label {
-	l.Component(PumpLabelVal)
-	return l
+	return l.Component(PumpLabelVal)
 }
 
-func (l Label) DMMaster() Label {
-	l.Component(DMMasterLabelVal)
-	return l
+// IsPump returns whether label is a Pump
+func (l Label) IsPump() bool {
+	return l[ComponentLabelKey] == PumpLabelVal
 }
 
-func (l Label) DMWorker() Label {
-	l.Component(DMWorkerLabelVal)
-	return l
+// DMMaster assigns dm-master to component key in label
+func (l Label) DMMaster() Label {
+	return l.Component(DMMasterLabelVal)
 }
 
+// IsDMMaster returns whether label is a DMMaster
 func (l Label) IsDMMaster() bool {
 	return l[ComponentLabelKey] == DMMasterLabelVal
 }
 
-func (l Label) IsDMWorker() bool {
-	return l[ComponentLabelKey] == DMWorkerLabelVal
+// DMWorker assigns dm-worker to component key in label
+func (l Label) DMWorker() Label {
+	return l.Component(DMWorkerLabelVal)
 }
 
-func (l Label) IsPump() bool {
-	return l[ComponentLabelKey] == PumpLabelVal
+// IsDMWorker returns whether label is a DMWorker
+func (l Label) IsDMWorker() bool {
+	return l[ComponentLabelKey] == DMWorkerLabelVal
 }
 
+// Monitor assigns monitor to component key in label
 func (l Label) Monitor() Label {
-	l.Component(TiDBMonitorVal)
-	return l
+	return l.Component(TiDBMonitorVal)
 }
 
+// IsMonitor returns whether label is a Monitor
 func (l Label) IsMonitor() bool {
 	return l[ComponentLabelKey] == TiDBMonitorVal
 }
 
 // Discovery assigns discovery to component key in label
 func (l Label) Discovery() Label {
-	l.Component(DiscoveryLabelVal)
-	return l
-}
-
-// IsPD returns whether label is a PD
-func (l Label) IsPD() bool {
-	return l[ComponentLabelKey] == PDLabelVal
+	return l.Component(DiscoveryLabelVal)
 }
 
 // TiDB assigns tidb to component key in label
 func (l Label) TiDB() Label {
-	l.Component(TiDBLabelVal)
-	return l
+	return l.Component(TiDBLabelVal)
+}
+
+// IsTiDB returns whether label is a TiDB
```

Comment:
Global comment: a TiDB, a DMWorker, a Pump is strange. We should either use `a TiDB cluster, a TiDB node` or `TiDB`.